### ISSUE TITLE
Unspaced Line Length Enforcement

### ISSUE THREAD

**Issue by t-rad679** [markdownlint/markdownlint#300]
## Describe the Enhancement:

As described in this [issue](https://github.com/markdownlint/markdownlint/issues/295), I noticed that markdownlint does not behave like many linters, in that its line length enforcement only triggers when there is a space after the line limit. I would like to have the option to wrap unbroken text to the next line unless the piece of text takes up the entirety of the line (accounting for indentation). For example:

I would like the following to trigger the rule:

```markdown
I would like to pretendthislineistoolong
```

While the following does not:

```markdown
pretendthistextistoolongbyitself
```

I would also like to ensure that indentation is preserved. For an example of this:

```markdown
*   Here is a list
    And also pretendthislineistoolong
```

Should trigger the rule, while the following does not:

```markdown
*   Here is a list
    pretendthislineistoolongasindented
```

Even if the line would not trigger the rule if it were not indented. Other, non-list indentations should also be accounted for similarly.

### Impacted Rules:

line-length

## Describe the Need:

It is much cleaner, in my opinion, to have unbroken text that is too long to be on its own line altogether, I find that the separation makes it easier to read. This is a common feature among linters; most that I have found operate this way: ESLint, Spotless, Checkstyle, and JetBrains built in linters.

## Curren t Alternative

Look through the code with regex search and manually break the lines. For a simple example, one could search for this pattern: ```/.{121}```

## Can We Help You Implement This?:

Lol time. Also, any collection of resources on Ruby that are related to this project and any resources you have on modifying rules in this project specifically. I have basically no experience with Ruby.

Thanks!

**Comment by jaymzh** [comment]
as mentioned I don't know we can reliably do *that* - there's a LOT of cases in which you might have leading space and I expect that t be SUPER brittle. But what I can offer, as I said is:

```
:allow_nonspaced_lines => true, :allow_nonspaced_overflow => true
```

**Comment by t-rad679** [comment]
Ok. Because of its nature as a more prose-oriented language, I could see this being the case, but how specifically do you see markdown as different from other languages in this way? This is the only case I've ever encountered where this feature was even a question for a linter. What cases other than lists and indented code blocks would have to be accounted for?

Can you also tell me the exact behavior you have in mind for the two options you're suggesting?
